test/core/derivatives/derivatives_test.py

```diff
@@ -58,14 +56,55 @@
 SUBSAMPLING_IDS = [f"subsampling={s}".replace(" ", "") for s in SUBSAMPLINGS]
 
 
+@pytest.mark.parametrize("subsampling", SUBSAMPLINGS, ids=SUBSAMPLING_IDS)
+@pytest.mark.parametrize(
+    "sum_batch", [True, False], ids=["sum_batch=True", "sum_batch=False"]
+)
+def test_param_jac_t_mat_prod(
+    problem: DerivativesTestProblem,
+    sum_batch: bool,
+    subsampling: List[int] or None,
+    request,
+) -> None:
+    """Test all parameter derivatives.
+
+    Args:
+        problem: test problem
+        sum_batch: whether to sum along batch axis
+        subsampling: subsampling indices
+        request: problem request
+    """
+    _skip_if_subsampling_conflict(problem, subsampling)
+    test_save_memory: bool = "Conv" in request.node.callspec.id
+
+    for param_str, _ in problem.module.named_parameters():
```

Comment:
Is there a way to decouple the tests for each parameter?

Reply:
I have the same objection and I didn't see a way to realize it. As a workaround, I added the `print(f"testing derivative wrt {param_str}")` line. If the test fails the user can still see which parameter failed.
If you see a way to decouple it would be much better.

Reply:
I have a solution that decouples the tests, but it not clean:
```python
PARAM_NAMES = ["weight", "bias", ...] # hard-coded
PARAM_IDS = [f"param_str={p}" for p in PARAM_NAMES]

pytest.mark.parametrize("param_str", PARAM_NAMES, ids=PARAM_IDS)
def test_param_mjp(param_str, problem, ...):

    check_all_params_in_PARAM_NAMES(problem) # make sure we don't accidentally forget params
    skip_if_has_no_param(param_str, problem) # skip if current problem does not have the param

    # test MJP for param_str
    ...
```

Reply:
**Update:** The skip condition is already on [development](https://github.com/f-dangel/backpack/blob/development/test/utils/skip_test.py#L77-L86).

Reply:
Ok. That would work.
I like the current version more, because skipping all these tests would result in many skipped test (4/6 would be skipped for most modules).

Reply:
Closing then.